### DISCUSSION TITLE
Question about linking text to the associated widgets

### DISCUSSION THREAD

**Discussion by sparvaneh** [pymupdf/PyMuPDF#987] [Q&A]
Hi,

Is there any example to show how to link widget values to the associated text? For example, I have a text "gender" and there are two radio buttons associated with "Male" and "Female". I am hoping to get advice on how to link widgets and text.

**Comment by JorjMcKie** [comment]
The question has several aspects:

* you can insert button widgets like any other field (see docu)
* fill widget property `button_caption` with desired text like "Male". This is shown in the widgets rectangle.
* fill optional `field_label` which may contain comments or help on what data should be used when filling out. Adult PDF viewers would display this, when the cursor hovers over the button.
* PyMuPDF **_does not support button groups_**, which e.g. switch every button off if one button of the group is switched on.
* Extra text like "Gender" would have to defined "manually" by using a normal text field.
